### ISSUE TITLE
Build fails: incompatible function pointer types initializing

### ISSUE THREAD

**Issue by yurivict** [MDAnalysis/mdanalysis#4589]
```
MDAnalysis/lib/formats/src/xdrfile.c:2535:3: error: incompatible function pointer types initializing 'unsigned int (*)(XDR *)' (aka 'unsigned int (*)(struct XDR *)') with an expression of type 'int64_t (XDR *)' (aka 'long (struct XDR *)') [-Wincompatible-function-pointer-types]
 2535 |                 xdrstdio_getpos,                /* get offset in the stream */
      |                 ^~~~~~~~~~~~~~~
MDAnalysis/lib/formats/src/xdrfile.c:2536:3: error: incompatible function pointer types initializing 'int (*)(XDR *, unsigned int)' (aka 'int (*)(struct XDR *, unsigned int)') with an expression of type 'int (XDR *, int64_t, int)' (aka 'int (struct XDR *, long, int)') [-Wincompatible-function-pointer-types]
 2536 |                 xdrstdio_setpos,                /* set offset in the stream */
      |                 ^~~~~~~~~~~~~~~
```

## Current version of MDAnalysis ##

- Which version are you using? 2.7.0
- Which version of Python (`python -V`)? 3.9
- Which operating system? FreeBSD 14.0
clang-17

**Comment by hmacdope** [comment]
Hi @yurivict this was fixed in #4408, try cloning a developer copy of the repo and installing the latest `develop`. Let me know how that goes?

**Comment by yurivict** [comment]
Ok, thanks,

I'll wait for the next release.

Yuri

**Comment by hmacdope** [comment]
I'll leave this open for anyone experiencing similar issues.

**Comment by yurivict** [comment]
People can only build this code successfully due to compiler bugs that allow invalid code.
You should probably consider to make a release with this fix soon.
(I myself applied a similar patch but others would likely suffer from this.)

**Comment by IAlibay** [comment]
> People can only build this code successfully due to compiler bugs that allow invalid code.

I just want to correct this:

1) GNU compiler compilations are not affected as far as we know
2) Nothing in our test suite fails nor do we have any evidence that these clang compilations lead to incorrect results

A release will come soon, although not until at the very least June.

**Comment by yurivict** [comment]
The GNU compiler version that you use is buggy. It is probably very old too.
Functions of incompatible signatures are allowed to be assigned, which should be an undefined behavior.
Recent clang-17 fails to compile it.

**Comment by IAlibay** [comment]
The compiler versions we use are those that come as default in GitHub actions runners and those that have been chosen as appropriate over conda forge.

In any case, because this discussion can cast doubt on the scientific validity of MDAnalysis, I again want to emphasize that we have no evidence that the validity of any results is in any way affected.

**Comment by orbeckst** [comment]
Given that this issue is a duplicate of #4397, which we already closed because of merged PR #4408, I would suggest to also close this issue.

I understand https://github.com/MDAnalysis/mdanalysis/issues/4589#issuecomment-2097098749 but I feel it's more consistent to close duplicates together with the original issues. @hmacdope I leave it to you to make the decision.

**Comment by orbeckst** [comment]
P.S.: Cool to see FreeBSD — gave me a reason to add a new OpSys-FreeBSD label :-)
